From 463f6d709c133a68a5f191b06bbc302ec8481348 Mon Sep 17 00:00:00 2001
From: Frank Li <Frank.Li@freescale.com>
Date: Wed, 14 Oct 2009 14:08:40 +0800
Subject: [PATCH] ENGR00117252 Fix iMX23 dotclk is double as expected value

Dotclk is double as expected value. This fix actually is workaround.

Signed-off-by: Frank Li <Frank.Li@freescale.com>
---
 arch/arm/plat-stmp3xxx/clock.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/arch/arm/plat-stmp3xxx/clock.c b/arch/arm/plat-stmp3xxx/clock.c
index 1de3109..6824182 100644
--- a/arch/arm/plat-stmp3xxx/clock.c
+++ b/arch/arm/plat-stmp3xxx/clock.c
@@ -219,6 +219,7 @@ static int lcdif_set_rate(struct clk *clk, u32 rate)
 	 * 108 * ns_cycle <= 875 * div
 	 */
 	u32 ns_cycle = 1000000 / rate;
+	ns_cycle *= 2; /* Fix calculate double frequency */
 	u32 div, reg_val;
 	u32 lowest_result = (u32) -1;
 	u32 lowest_div = 0, lowest_fracdiv = 0;
-- 
1.5.4.4

